### PR TITLE
[REVIEW] Replace use of assert_frame_equal in tests with assert_eq

### PR DESCRIPTION
**Pull request by shwina**
Since the logic for comparing cudf/Pandas objects is in the `assert_eq` utility function, we should aim to use that wherever possible.

cc: @brandon-b-miller

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

**Comment by shwina** [comment]
Converting to draft as this is still WIP (need to replace use of `assert_series_equal` and `assert_index_equal`)

**Comment by brandon-b-miller** [comment]
I also noticed a few things like `np.testing.assert_approx_equal` in `test_stats.py`.

**Comment by shwina** [comment]
@brandon-b-miller since those are comparing against NumPy arrays (which don't have the concept of `NA`), I think we can leave as is. This is ready for another look from my perspective

File: CHANGELOG.md [hunk 1]
```diff
@@ -128,6 +128,8 @@
 - PR #5709 Update java build to help cu-spacial with java bindings
 - PR #5713 Remove old NVTX utilities
 - PR #5726 Replace use of `assert_frame_equal` in tests with `assert_eq`
+- PR #5720 Replace owning raw pointers with std::unique_ptr
```
Comment (kkraus14) [inline CHANGELOG.md:131]: Needs merge to resolve this duplicate changelog line?

**Pull request merged**